src/main/java/org/mockito/MockitoAnnotations.java

```diff
@@ -37,8 +37,14 @@
  *
  *   public class SampleBaseTestCase {
  *
+ *       private AutoCloseable closeable;
+ *
  *       &#064;Before public void initMocks() {
- *           MockitoAnnotations.initMocks(this);
+ *           closeable = MockitoAnnotations.initMocks(this);
```

Comment:
This is potentially a breaking change, if compilations downstream require the result of any `AutoCloseable` to be used. I would have to check for example internally if that is the case. Can we maybe separate this out into a separate method called `MockitoAnnotations.initWithStaticMocks`? That would make it clearer that a particular test is allowed to use static mocks. By using a separate method, users can write static analyses that verify that `.close` is called.

Reply:
I thought about this, too. It is source compatible but not binary compatible. However, I do not think that this is a very commonly used API by derivative tools and direct users do not have to apply any changes until using static mocks in their custom tooling what likely requires them to revisit this API. In a way, explicit life-cycles are a new concept in Mockito that get introduced by this change, it will require some upstream adoption and in such a case, I think that breakage is the most explicit way to communicate this to users; I always prefer this over silent or runtime failure. I suggest therefore to make this visible in the release notes and by the versioning scheme via releasing a new feature version when merging this.